docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+            model.compile(loss=losses.BinaryCrossentropy(from_logits=True),
+                optimizer='adam',
+                metrics=tf.metrics.BinaryAccuracy(threshold=0.0))
+            logger.info("beginning training...")
+            history = model.fit(train_ds, validation_data=val_ds,
+                    epochs=epochs)
+            mp_queue.put(history)
+            logger.info("Saving...")
+            model.save(model_save_loc, save_format='tf', overwrite=True)
+        except Exception as e:
+            logger.error(f"In precompile {e}")
+
+        try:
+            logger.info(f"Training complete!")
+            export_model = tf.keras.Sequential([
+                vectorize_layer,
+                model,
+                layers.Activation('sigmoid')
+                ])
+            logger.info(f"Compiling new model...")
+            export_model.compile(
+                loss=losses.BinaryCrossentropy(from_logits=False), optimizer="adam", metrics=['accuracy']
+            )
+            loss, accuracy = model.evaluate(test_ds)
+            logger.info(f"Evaluation Results: \n {str(loss)} \n {str(accuracy)}")
+            export_model.save(f"{model_save_loc}-exported", save_format="tf", overwrite=True)
+        except Exception as e:
+            logger.error(f"ERR in TRAIING \n {e}")
+            raise e
+```
+
+While this method may be intimidating, don't let it scare you! It is simply the core code of our companion piece, gently refactored to fit into our tool's (and the PythonSdk's) runtime and lifecycle.
+
+In this method we:
+  * load our dataset `tf.data.Dataset.save's sibling
+  * create and adapt our `vectorize_layer: tensorflow.keras.layers.TextVectorization` as recommended by TensorFlow
+  * cache our datasets (now loading data as `Datasets` begins to pay off!)
+  * init, compile, evaluate, and save our base model.]
+  * generate  data for our previews
+
+You may have noticed we use a decorator here as well `static method`.
+We do this to allow us to call this using `Multiprocessing`  to train our model without blocking our Python sdk service process. We also add in a multiprocessing queue to quickly retrieve the results and send them to the front end before ending.
+Lastly, note that we catch a generalized exception and explicitly raise it for this method. Doing that allows us to throw in a controlled way such that the Designer can report but log it _before_ the process terminates, as we discussed early on in this section of the guide.
+
+As you might have noticed, we currently have a non-trivial amount of arguments.
+Since we collect these via IPC (**Read:** InterProcess Communication: N processes running separately; think to Excel and tablue with a linked spreadsheet.), these will also be more difficult to keep in sync.
+As such, we want a function to extract these that we can wrap in a try/catch block to explicitly capture and report `KeyErrors` or similarly common exceptions in this context.
```

Comment:
TW review:

As you might have noticed, we currently have a non-trivial amount of arguments. Since we collect these via IPC (Read: InterProcess Communication: N processes running separately. Think to Excel and table with a linked spreadsheet.), these are also more difficult to keep in sync. As such, we want a function to extract these that we can wrap in a try/catch block to explicitly capture and report `KeyErrors` (or similarly common exceptions in this context).

Reply:
Resolution Note: removed excel example, re-reading I feel it makes an obvious issue unclear haha.